api/test/integration/test_media_integration.py

```diff
@@ -329,6 +329,57 @@ def test_detail_view_for_invalid_uuids_returns_not_found(
     assert res.status_code == 404
 
 
+def test_search_with_only_valid_sources_produces_no_warning(media_type, api_client):
```

Comment:
Nit: This is specialised case of the next test function which could be parametrized (in terms of invalid sources being an empty or populated set) to include this scenario.

Reply:
I thought of that too, but I liked having separate and descriptive names for them, rather than doing an if/else in the test to do the assertions for each one.